modules/ml/src/main/java/org/apache/ignite/ml/math/functions/Functions.java

```diff
@@ -215,15 +215,73 @@
     }
 
     /**
-     * Curry bifunction.
+     * Curry bi-function.
      *
-     * @param f Bifunction to curry.
+     * @param f Bi-function to curry.
      * @param <A> Type of first argument of {@code f}.
      * @param <B> Type of second argument of {@code f}.
      * @param <C> Return type of {@code f}.
-     * @return Curried bifunction.
+     * @return Curried bi-function.
      */
     public static <A, B, C> IgniteCurriedBiFunction<A, B, C> curry(BiFunction<A, B, C> f) {
         return a -> b -> f.apply(a, b);
     }
+
+    /**
+     * Transform bi-function of the form (a, b) -> c into a function of form a -> (b -> c).
+     *
+     * @param f Function to be curried.
+     * @param <A> Type of first argument of function to be transformed.
+     * @param <B> Type of second argument of function to be transformed.
+     * @param <C> Type of third argument of function to be transformed.
+     * @return Curried bi-function.
+     */
+    public static <A, B, C> IgniteCurriedBiFunction<A, B, C> curry(IgniteBiFunction<A, B, C> f) {
```

Comment:
As far as I understand `curry` is an operator that accepts a function with N arguments and M values and produces a function with N-M arguments. So, [currying](https://en.wikipedia.org/wiki/Currying#Definition) in this context is an action of fixing subset of function arguments and the result of the currying should be the function with N-M arguments.
In the current implementation I see that the `curry` method returns the function with the same number of elements as in the accepted function. For me it looks like it should be `IgniteCurryingFunction`, not `Curried`, shouldn't it? 
Or even better, the `curry` method can accept a function and arguments and return actually curried function (just an `IgniteFunction` with one argument in this case) to avoid extra function wrappers.

```
public static <A, B, C> IgniteFunction<B, C> curry(IgniteBiFunction<A, B, C> f, A a) {
  return b -> f.apply(a, b);
}
```

What do you think?